src/OpenSearch/Endpoints/AbstractEndpoint.php

```diff
@@ -22,26 +22,27 @@
 namespace OpenSearch\Endpoints;
 
 use OpenSearch\Common\Exceptions\UnexpectedValueException;
+use OpenSearch\EndpointInterface;
 use OpenSearch\Serializers\SerializerInterface;
 
 use function array_filter;
 
-abstract class AbstractEndpoint
+abstract class AbstractEndpoint implements EndpointInterface
 {
     /**
      * @var array
      */
-    protected $params = [];
+    protected array $params = [];
```

Comment:
add native types on protected properties is a BC break in case those properties are redefined in a child class